packages/babel-parser/src/tokenizer/index.js

```diff
@@ -123,13 +127,21 @@ export default class Tokenizer extends LocationParser {
     this.isLookahead = false;
   }
 
+  pushToken(token: Token | N.Comment) {
+    // Pop out invalid tokens trapped by try-catch parsing.
+    // Those parsing branches are mainly created by typescript and flow plugins.
+    this.tokens.length = this.state.tokensLength;
+    this.tokens.push(token);
```

Comment:
I know the `array#splice` is always hacky, but since it is a frequent operation, can you simplify this to
```js
this.tokens.splice(this.state.tokensLength, Infinity, token)
```
It should be marginally faster since the engine don't have to reallocate memory if `this.tokens.length > this.state.tokensLength`

Reply:
BTW, there would not be a situation for `this.tokens.length > this.state.tokensLength`.